CoreFoundation/Base.subproj/CFUtilities.c

```diff
@@ -1265,7 +1277,7 @@ CFDictionaryRef __CFGetEnvironment() {
         extern char **environ;
         char **envp = environ;
 #elif DEPLOYMENT_TARGET_LINUX
-#ifndef environ
+#if !defined(environ) && !defined(__ANDROID__)
```

Comment:
envrion is a C symbol not a #define for the preprocessor so you can’t test for whether it exists with a #ifdef.